### ISSUE TITLE
Issue about encode 

### ISSUE THREAD

**Issue by ChenD1998** [kongchen1992/deep-nrsfm#8]
I read data from **23.train** and then rewrite it into **25.train** as follow:
```
def _parse_function(example_proto):
    """Parses raw bytes into tensors."""
    features = {
        "points3d_raw": tf.FixedLenFeature((), tf.string, default_value=""),
        "points2d_raw": tf.FixedLenFeature((), tf.string, default_value=""),
    }
    parsed_features = tf.parse_single_example(example_proto, features)
    output_features = {
        "points3d": tf.reshape(
            tf.decode_raw(parsed_features["points3d_raw"], tf.float32),
            [NUM_POINTS, 3],
        ),
        "points2d": tf.reshape(
            tf.decode_raw(parsed_features["points2d_raw"], tf.float32),
            [NUM_POINTS, 2],
        ),
    }
    # Returns a tuple (features, labels)
    # return output_features, 0
    return output_features

def write():
    output_record_dir = "dataset/tfrecords/25.train"
    writer = tf.python_io.TFRecordWriter(output_record_dir)

    filename="dataset/tfrecords/23.train"

    dataset = tf.data.TFRecordDataset(filename)
    dataset = dataset.map(_parse_function)
    temp = dataset.make_one_shot_iterator().get_next()

    with tf.Session() as sess:
        while True:
            try:
                my_point2d = sess.run(temp['points2d'])
                my_point3d = sess.run(temp['points3d'])

                example = tf.train.Example(features=tf.train.Features(feature={
                        "points3d_raw": tf.train.Feature(bytes_list=tf.train.BytesList(
                            value=[my_point3d.tostring()])),
                        "points2d_raw": tf.train.Feature(bytes_list=tf.train.BytesList(
                            value=[my_point2d.tostring()])),
                }))

                writer.write(example.SerializeToString())
            except tf.errors.OutOfRangeError:
                break
    writer.close()
```
There are no differences between data from **23.train** and mine(**25.train**). 
 ```
    filename = "dataset/tfrecords/23.train"
    file2 = "dataset/tfrecords/25.train"

    dataset = tf.data.TFRecordDataset(filename)
    dataset = dataset.map(_parse_function)
    dataset2 = tf.data.TFRecordDataset(file2)
    dataset2 = dataset2.map(_parse_function)

    temp = dataset.make_one_shot_iterator().get_next()
    temp2 = dataset2.make_one_shot_iterator().get_next()

    with tf.Session():
        while True:
            try:
                my_point2d = temp['points2d'].eval()            
                t_point2d = temp2['points2d'].eval()                            
                # print diff
                print(my_point2d-t_point2d)
                # output is zero matrix
                break
            except tf.errors.OutOfRangeError:
                break
```
After training with the same hparams as you do, the loss of my model con- verges to around 0.02. 
But it performs bad ,which the average of error_abs is 0.15, which is much larger than your 0.04.
I just read data and rewrite it my way. 
So I want to know how do you encode data to **23.train**. Thank you!
(I find no more helpful infomation on CMU Graphics Lab Motion Capture Database, and it seems to be no link to download data)

**Comment by kongchen1992** [comment]
I would recommend you to 1) directly read data from 23.train and 25.train and check if they are exactly the same.  2) apply the pretrained model on both data and check the performance.  If there is no bug when rewriting data, I would think these two data should result in the same evaluation error.  Let me know what you got.

**Comment by kongchen1992** [comment]
Closed this issue due to no-reply for a period of time.
